Fix several CSS elements and bugs for Sphinx 5

choldgraf opened the pull request:
This PR touches the CSS in several different places, with the goal of making sure it works with Sphinx 5 / docutils .18+. It also fixes a few bugs in the process as well as a few DRY improvements by re-using SCSS variables.

This fixes all of the things things mentioned in the issue below, closes #817 .

- #817

In addition, there were a few other things I fixed up along the way. They were noticeable when I was testing locally with docutils .18/sphinx 5, but I'm not sure exactly which were due to the version change and which were just things to improve in general, so it's kind-of a mishmash of stuff.

- Re-use SCSS variables in a few places (e.g. for border radius) and HTML variables in others (e.g. header/caption weight)
- Improve styling of the sidebar in general to be similar to our code caption blocks, and make other elements lay out properly when next to a sidebar.
- Removed the box-shadow behind our code blocks, since it wasn't clear what it was accomplishing other than making our shadow a bit wider (but this was invisible in dark mode, and only 1px in light mode).
- Cleaning up our sphinx-copybutton and sphinx-togglebutton code a bit for their latest releases (removes most of the togglebutton special-casing).
- Getting our figure captions to center properly

choldgraf commented:
I'd appreciate a review from somebody on this, at least at the level of how the visual style looks in the end result. It would be nice to get support for Sphinx 5 in the next release!

drammock commented:
working on a thorough review. one thing that looks odd so far: the left border color on this code cell:

![Screenshot 2022-07-26 at 16-47-24 PyData Library Styles — PyData Theme 0 10 0rc1 documentation](https://user-images.githubusercontent.com/1810515/181118396-04d3d745-38f9-42e1-95d6-2bb6af899be4.png)

drammock reviewed (approved):
looks good!  I didn't spot anything odd-looking in the rendered docs that isn't already that way on `main` / devdocs. Code changes look reasonable, just a couple small comments.

choldgraf commented on src/pydata_sphinx_theme/assets/styles/content/_footnotes.scss:
```diff
@@ -13,3 +13,12 @@ a.footnote-reference {
 }
+
+// Docutils 0.18 uses an `aside.footnote` container with differen internal structure
+aside.footnote {
+  margin-bottom: 1rem;
+  span.label,
+  span.backrefs {
+    font-weight: bold;
+  }
+}
```
```suggestion
aside.footnote {
  margin-bottom: .5rem;
  &:last-child {
    margin-bottom: 1rem;
  }
  span.label,
  span.backrefs {
    font-weight: bold;
  }
}
```

choldgraf commented:
thanks @drammock ! I implemented both suggestions 👍 I like your footnote spacing better :-)

choldgraf commented on src/pydata_sphinx_theme/assets/styles/content/_footnotes.scss:
```diff
@@ -13,3 +13,15 @@ a.footnote-reference {
+
+// Docutils 0.18 uses an `aside.footnote` container with different internal structure
+aside.footnote {
+  margin-bottom: .5rem;
```
```suggestion
  margin-bottom: 0.5rem;
```

choldgraf commented:
### Ready for review / merge

I just rebased this one on `main` and fixed a few bugs that I noticed as a result. I think it'd be helpful if somebody took another look at the resulting CSS in our Read the Docs build to see if there are any obvious CSS bugs introduced.

If we don't notice any, I suggest that we merge this in and make an `rc2`. @jarrodmillman would you be willing to help with that? :-)

tupui commented:
I tested this branch with SciPy and it works great. Only issue I still get is the empty sidebar when there is no element to show. 

<img width="273" alt="Screenshot 2022-07-28 at 00 15 24" src="https://user-images.githubusercontent.com/23188539/181444254-c3d2ca92-5238-46cc-9a78-068a798b0317.png">
 
It's more a problem on small screens as the right panel will be present and empty.
<img width="557" alt="Screenshot 2022-07-28 at 00 16 21" src="https://user-images.githubusercontent.com/23188539/181444722-257381a3-d9ed-4864-b171-b9f83eaeb468.png">
<img width="562" alt="Screenshot 2022-07-28 at 00 16 32" src="https://user-images.githubusercontent.com/23188539/181444728-9f0963be-d2ea-41b7-a7b7-dd0564cf9a8b.png">

Minor things I saw. Some left-right padding on a button got removed and on Sphinx-Design cards it was adding some horizontal lines around the body.

<img width="449" alt="Screenshot 2022-07-28 at 00 35 36" src="https://user-images.githubusercontent.com/23188539/181448608-04110522-a755-4824-9f49-51c6563bc009.png">
<img width="184" alt="Screenshot 2022-07-28 at 00 35 59" src="https://user-images.githubusercontent.com/23188539/181448618-3f24358d-8ccb-4929-916b-01bacb12c5df.png">

tupui commented:
And something weird is happening upon resizing before it goes to the mobile mode
<img width="1061" alt="Screenshot 2022-07-28 at 00 45 22" src="https://user-images.githubusercontent.com/23188539/181450432-02bbb97e-bfbe-47ed-b1cd-2e70eba0429e.png">

choldgraf commented:
> And something weird is happening upon resizing before it goes to the mobile mode

That's what happens when it runs out of horizontal space...not sure how to deal with that yet haha. You could try `navbar_align: left` to align the navbar items to the left, just by the logo. That should create more space

drammock commented:
> And something weird is happening upon resizing before it goes to the mobile mode

This is reported in #845, further discussion of this problem can go there

choldgraf commented:
By the way - I welcome others to push changes to this PR branch, I am really busy with a grant deadline for the next few days and would like to get a new release out soon if possible because there are many nice new changes to add!

choldgraf commented:
OK I think that I fixed the sidebar one in https://github.com/pydata/pydata-sphinx-theme/pull/849

Want to try that out @tupui ? 

I'm going to merge this, and cut a new RC so that we can have people test out

tupui commented:
@choldgraf I just tried `main` and I still have in the output this defined `bd-sidebar-secondary bd-toc` although the TOC is empty.

choldgraf commented:
Can you provide a link or open up an issue with a reproducible set of steps? I can't figure out how to reproduce on my side

tupui commented:
@choldgraf you can use this PR: https://github.com/scipy/scipy/pull/16660

To build the SciPy and the doc (build should take around 3 mins with 6 cores, then 5 mins for the doc itself):

```
mamba create env -f environment.yml
python dev.py doc -j 6  # number of cores
```

Thanks for looking into this Chris! Let me know if you want me to try something to help.

choldgraf commented:
Hmmm you don't have a Read the Docs PR preview to use? I'm sorry but I don't have time to download and build the scipy docs locally, I'm on vacation right now :-/

tupui commented:
Oh sure no worries. If it's a build that you need, I can update the current PR I have so that it builds with the RC. All that can wait until you are back. FYI SciPy 1.9 got released this WE.

choldgraf commented:
Ok sounds good - I think we should open a separate issue to track this instead of reusing this PR. Can you do that and provide a link to the reproducible example there?

Do you think this should block the next release if the theme?

tupui commented:
Sure, I will open a new issue. What I am seeing now that is more of an issue for the next release is that build time significantly increased. I noticed this as well locally but thought it was a hiccup at first. But on the CI it seems to be consistently happening (timeout, not even finishing).

choldgraf commented:
Can you please open new issues for anything? 🙂

QuLogic commented on docs/user_guide/theme-elements.md:
````diff
@@ -118,6 +118,17 @@ I was made with the `{admonition}` directive and a `sidebar` class.
 I was made with the `{sidebar}` directive.
 ```
 
+## Footnotes
+
+Here's one footnote[^1] and another footnote [^2] and a named footenote[^named], symbol [^*].
+
+[^1]: Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar.
+[^2]: Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar.
+[^named]: Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar.
+[^*]: Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar. Foo bar foo bar.
+
+## Version changes
+
````
This doesn't look correct? It is a whole section inserted into the middle of another one (Admonition sidebars). It also restores the `Version changes` heading that was removed in https://github.com/pydata/pydata-sphinx-theme/pull/836/files#diff-0607a043e4091f4e22ccc62684762b4a6b00d03e638ff096277e7f4dde9cba77L97, but none of its section content.
